test/TritonGPU/amd/amd-range-analysis.mlir

```diff
@@ -929,3 +929,22 @@ module attributes {"ttg.num-warps" = 4 : i32} {
     tt.return %11 : tensor<1024xf32>
   }
 }
+
+// -----
+
+// CHECK-LABEL:   tt.func @DynamicKBound
+module attributes {"ttg.num-warps" = 4 : i32} {
+  tt.func @DynamicKBound(%K: i32) {
+    // expected-remark@+2 {{unsigned : [1024, 1024] signed : [1024, 1024]}}
+    // expected-remark@+1 {{non-neg}}
+    %c1024_i32 = arith.constant 1024 : i32
+    // expected-remark@+2 {{unsigned : [128, 128] signed : [128, 128]}}
+    // expected-remark@+1 {{non-neg}}
+    %c128 = arith.constant 128 : i32
+    %cmp = arith.cmpi sle, %K, %c128 : i32
+    llvm.intr.assume %cmp : i1
+    // expected-remark@+1 {{unsigned : [-1, -1] signed : [-1, -1]}}
```

Comment:
note this is `sext(0x1)` because we still have not caught up to https://github.com/llvm/llvm-project/pull/127888 which fixed the `dbg` print of the ranges.